core/contractsdk/cpp/example/group_chain.cc

```diff
@@ -0,0 +1,168 @@
+#include "xchain/xchain.h"
+
+const std::string nodeBucket = "NODE";
+const std::string chainBucket = "CHAIN";
+
+const std::string endingSeparator = "\x01";
+
+class GroupChainBasic {
+public:
+    virtual void initialize() = 0;
+    virtual void listNode() = 0;
+    virtual void addNode() = 0;
+    virtual void delNode() = 0;
+    virtual void getNode() = 0;
+    virtual void changeIP() = 0;
+    virtual void getChain() = 0;
+    virtual void addChain() = 0;
+    virtual void delChain() = 0;
+    virtual void listChain() = 0;
+};
+
+class GroupChain : public GroupChainBasic, public xchain::Contract {
+public:
+    void initialize() {
+        xchain::Context* ctx = this->context();
+        ctx->ok("initialize success");
+    }
+    void listNode() {
+        xchain::Context* ctx = this->context();
+        std::string key = nodeBucket + ctx->arg("bcname") + endingSeparator;
+        std::unique_ptr<xchain::Iterator> iter = ctx->new_iterator(key, key + "～");
+        std::string result;
+        while (iter->next()) {
+            std::pair<std::string, std::string> res;
+            iter->get(&res);
+            std::string target = res.first;
+            int offset = key.length() + 1;
+            int length = target.length() - offset;
+            result += target.substr(offset, length) + endingSeparator;
+        }   
+        ctx->ok(result);
+    }
+    void addNode() {
+        xchain::Context* ctx = this->context();
+        std::string key = nodeBucket + ctx->arg("bcname") + endingSeparator + ctx->arg("ip");
```

Comment:
arg("bcname") should not be "xuper", may check here

Reply:
corrected.